packages/rainbowkit/src/wallets/walletConnectors/coinbaseWallet/coinbaseWallet.ts

```diff
@@ -8,13 +8,55 @@ export interface CoinbaseWalletOptions {
   chains: Chain[];
 }
 
+declare global {
+  interface Window {
+    coinbaseWalletExtension: Window['ethereum'];
+  }
+}
+
+function getCoinbaseWalletInjectedProvider(): Window['ethereum'] {
+  const isCoinbaseWallet = (
+    ethereum: NonNullable<Window['ethereum']> | any,
+  ) => {
+    // Identify if Coinbase Wallet injected provider is present.
+    const coinbaseWallet = !!ethereum.isCoinbaseWallet;
+
+    return coinbaseWallet;
+  };
```

Comment:
Just made sure to add proper check for coinbase wallet since coinbase wallet can be replaced by another injected provider. This way we can see if it's installed properly or not.